Sources/SWBMacro/MacroValueAssignmentTable.swift

```diff
@@ -396,6 +423,37 @@ public final class MacroValueAssignment: Serializable, CustomStringConvertible,
     }
 }
 
+public struct MacroValueAssignmentLocation: Serializable, Sendable {
+    public let path: Path
```

Comment:
I wonder if we're going to end up wanting to intern these strings somewhere to cut down on memory usage in projects with large settings tables

Reply:
Yah, that's a good point. I think settings already track paths of XCConfigs for invalidation purposes, maybe we can re-use that instead of duplicating the paths per macro like I am doing here.

Reply:
Updated to intern the paths at the macro assignment table level. 

We could go one step further and share storage here with settings since a fully constructed settings object also carries a list of its macro config files already, but that doesn't seem necessary and could be revisited if it turns out to be relevant.